### PR TITLE
feat(tha-766): integrate coingecko metrics

### PR DESCRIPTION
**Pull request by wesgraham**
CoinGecko requires metadata for its required endpoints for integration that we have presently missing from our indexer. This PR does 2 things: 
1. adds a pool liquidity gauge - which tracks total pool liquidity for the mandatory response of the /tickers endpoint
2. unifies `poolTag` and `poolType` for swap and liquidity events - we need a consistent tag to be used across these events to be able to fetch relevant metadata for coinGecko's /tickers endpoint

Reference: https://docs.google.com/document/d/1v27QFoQq1SKT3Priq3aqPgB70Xd_PnDzbOCiuoCyixw/edit

### CODE REVIEW

File: src/processors/base_pool.ts [hunk 1]
```diff
@@ -21,14 +21,15 @@ const volOptions = {
...
+const liquidityGauge = Gauge.register("pool_liquidity_usd", commonOptions);
 const volumeGauge = Gauge.register("pool_volume_usd", volOptions);
 const feeGauge = Gauge.register("pool_swap_fee_usd", volOptions);
 
 export async function onEventSwapEvent(
   ctx: AptosContext,
   type: "weighted" | "stable",
   coins: string[],
-  poolTag: string,
+  poolType: string,
```
Comment (wesgraham) [inline src/processors/base_pool.ts:32]: note: i am adding a `DO NOT MERGE` label to this PR as I have yet to still verify if this change will break queries downstream in our dapp. Once I verify i will remove the label

File: src/processors/base_pool.ts [hunk 2]
```diff
@@ -139,9 +141,12 @@ export async function onEventLiquidityEvent(
   ctx.eventLogger.emit("liquidity", {
     distinctId: ctx.transaction.sender,
     liquidityEventType,
-    pool,
+    poolType,
     value: usdValue.isNaN() ? new BigDecimal(0) : usdValue,
   });
+
+  const relativeUsdValue = liquidityEventType === "Add" ? usdValue : usdValue.times(-1);
+  liquidityGauge.record(ctx, relativeUsdValue, { poolType })
```
Comment (wesgraham) [inline src/processors/base_pool.ts:149]: liquidityGauge will maintain a running tally of the liquidity of the poolType

File: src/processors/base_pool.ts (hunk 1 above)
Comment (wesgraham) [reply src/processors/base_pool.ts:32]: verified!

File: src/processors/base_pool.ts [hunk 3]
```diff
@@ -136,12 +138,16 @@ export async function onEventLiquidityEvent(
     .map((amount, i) => amount.multipliedBy(actualCoinPrices[i]))
     .reduce((acc, e) => acc.plus(e), new BigDecimal(0));
 
+  let value = usdValue.isNaN() ? new BigDecimal(0) : usdValue;
```
Comment (0xbe1) [inline src/processors/base_pool.ts:141]: let value => const value

**Review by 0xbe1 — APPROVED** [review]
lgtm

File: src/processors/base_pool.ts (hunk 2 above)
Comment (0xbe1) [reply src/processors/base_pool.ts:150]: is this needed by coingecko?

Comment (wesgraham) [reply src/processors/base_pool.ts:150]: yes - coinGecko's ticker API requires `liquidity_in_usd` for reach pair. We loop up the latest price for the pairTag, and whatever pool corresponds to that price is used as the returned poolId. Then we return the liquidity of that pool as well in the request

**Pull request merged**